Networking/Extensions/DecodedExtension.swift

```diff
@@ -29,6 +29,17 @@ public extension Decoded {
     
 }
 
+public extension Decodable where Self: RawRepresentable, Self.RawValue == String {
+    
+    static func decode(_ json: JSON) -> Decoded<Self> {
+        switch json {
+        case let .string(name): return .fromOptional(Self(rawValue: name))
+        default: return .failure(Argo.DecodeError.custom("Invalid enum value"))
```

Comment:
I would give a little bit more information. Maybe something like `"Invalid \(Self.self) enum value"` if it works. So that the developer knows at which enum it failed.